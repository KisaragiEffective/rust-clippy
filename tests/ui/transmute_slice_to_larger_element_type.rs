#![allow(unused)]
#![warn(clippy::transmute_slice_to_larger_element_type)]

fn i8_slice_to_i32_slice() {
    let i8_slice: &[i8] = &[1i8, 2, 3, 4];
    let i32_slice: &[i32] = unsafe { std::mem::transmute(i8_slice) };
}

fn main() {}
