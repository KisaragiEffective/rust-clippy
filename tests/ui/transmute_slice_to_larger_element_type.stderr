error: transmute from `&[i8]` to `&[i32]` results in undefined behavior
  --> $DIR/transmute_slice_to_larger_element_type.rs:6:38
   |
LL |     let i32_slice: &[i32] = unsafe { std::mem::transmute(i8_slice) };
   |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this transmute leads out-of-bounds read
   = note: `-D clippy::transmute-slice-to-larger-element-type` implied by `-D warnings`
help: try
   |
LL |     let i32_slice: &[i32] = unsafe { i8_slice.iter().map(|item| unsafe { std::mem::transmute(item) }).collect::<Vec<_>>().to_slice() };
   |                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
LL |     let i32_slice: &[i32] = unsafe { std::slice::align_to::<i32>(i8_slice).1 };
   |                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error: aborting due to previous error

